perception/util/visualizer/traffic_light_visualization/src/nodelet.cpp

```diff
@@ -18,43 +18,52 @@
 
 namespace traffic_light
 {
-void TrafficLightRoiVisualizerNodelet::onInit()
-{
-  nh_ = getNodeHandle();
-  pnh_ = getPrivateNodeHandle();
-  image_transport_.reset(new image_transport::ImageTransport(pnh_));
-
-  pnh_.param<bool>("enable_fine_detection", enable_fine_detection_, false);
+TrafficLightRoiVisualizerNodelet::TrafficLightRoiVisualizerNodelet(const rclcpp::NodeOptions & options)
+: Node("traffic_light_roi_visualizer_node", options)
+{  
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+  using std::placeholders::_3;
+  enable_fine_detection_ = this->declare_parameter("enable_fine_detection", false);
 
   if (enable_fine_detection_) {
     sync_with_rough_roi_.reset(new SyncWithRoughRoi(SyncPolicyWithRoughRoi(10), image_sub_, roi_sub_,  rough_roi_sub_));
-    sync_with_rough_roi_->registerCallback(boost::bind(&TrafficLightRoiVisualizerNodelet::imageRoughRoiCallback, this, _1, _2, _3));
+    sync_with_rough_roi_->registerCallback(std::bind(&TrafficLightRoiVisualizerNodelet::imageRoughRoiCallback, this, _1, _2, _3));
   } else {
     sync_.reset(new Sync(SyncPolicy(10), image_sub_, roi_sub_));
-    sync_->registerCallback(boost::bind(&TrafficLightRoiVisualizerNodelet::imageRoiCallback, this, _1, _2));
+    sync_->registerCallback(std::bind(&TrafficLightRoiVisualizerNodelet::imageRoiCallback, this, _1, _2));
   }
-  image_transport::SubscriberStatusCallback connect_cb = boost::bind(&TrafficLightRoiVisualizerNodelet::connectCb, this);
-  std::lock_guard<std::mutex> lock(connect_mutex_);
-  image_pub_ = image_transport_->advertise("output/image", 1, connect_cb, connect_cb);
+  
+  auto timer_callback = std::bind(&TrafficLightRoiVisualizerNodelet::connectCb, this);
+  const auto period_s = 0.1;
+  const auto period_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(period_s));
+  timer_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_callback)>>(
+    this->get_clock(), period_ns, std::move(timer_callback),
+    this->get_node_base_interface()->get_context());
+  this->get_node_timers_interface()->add_timer(timer_, nullptr);
```

Comment:
Is this timer needed because SubscriberStatusCallback is not available in ROS2 and/or Foxy?

Reply:
indeed, this is done as in #70

Reply:
Thanks for linking the existing package!